### PR TITLE
Add integration testing using playwright

### LINKED ISSUE #1596: Add integration testing
We only have Python-level unit testing at the moment, which means we have nothing in place to test things like "do recaptchas load", "are there client-side errors on any of the pages we need", "does page navigation work", etc. and it's hit us _hard_ in the last few weeks.

I would recommend we add `playwright` to this repo with a slew of integration tests to make sure all critical page flows are covered.

### PR DESCRIPTION
**Pull request by Pomax**
This adds the same kind of integration testing that we have on the foundation repo to the donate repo, so we can do actual page testing.

Note that the fact that this runs in CI and passes is pretty much "this works", but to run this locally the instructions are basically the same as for the foundation repo. 

Closes #1596

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,10 +49,10 @@
     "babel-polyfill": "^6.26.0",
     "braintree-web": "^3.62.2",
     "intersection-observer": "^0.10.0",
-    "node-sass": "^4.14.1",
```
Comment (Pomax) [inline package.json:49]: node-sass is both obsolete and incompatible with modern Node. Replaced with the proper `sass` package

File: tests/wait-for-images.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Playwright has no built-in way to wait for all on-page images
```
Comment (Pomax) [inline tests/wait-for-images.js:2]: same file as on the foundation repo

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -61,3 +61,54 @@ jobs:
...
+      X_FRAME_OPTIONS: DENY
+      XSS_PROTECTION: True
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - uses: actions/setup-node@v2
+      with:
+        node-version: 14
```
Comment (danielfmiranda) [inline .github/workflows/continuous-integration.yml:97]:
```suggestion
        node-version: 14
        cache: 'npm'
```

File: .github/workflows/continuous-integration.yml [hunk 4]
```diff
@@ -61,3 +61,54 @@ jobs:
...
+      SLACK_WEBHOOK_PONTOON: ${{ secrets.SLACK_WEBHOOK_PONTOON }}
+      SSL_REDIRECT: False
+      USE_S3: False
+      X_FRAME_OPTIONS: DENY
+      XSS_PROTECTION: True
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
```
Comment (danielfmiranda) [inline .github/workflows/continuous-integration.yml:94]:
```suggestion
        python-version: 3.9
        cache: 'pip'
```

**Review by danielfmiranda — APPROVED** [review]
Hi Pomax, this looks good to me so I am approving! 👍 

However I did add two suggested comments just in case you wanted to cache the CI runs like we do in https://github.com/mozilla/foundation.mozilla.org/pull/8100

**Comment by Pomax** [comment]
good call

**Pull request merged**